src/main.rs

```diff
@@ -116,3 +116,123 @@ fn main() {
+
+                sess.userauth_password("admin", "cloud123")
+                    .map_err(|_| Error::Authentication)?;
+                assert!(sess.authenticated());
+
+                let mut channel = sess.channel_session().map_err(|_| Error::Command)?;
+                channel.exec(command).map_err(|_| Error::Command)?;
+
+                // Intentionally ignore these results here as their failure
+                // does not precipitate a repeat
+                let _ = channel.read_to_string(&mut s);
+                let _ = channel.close();
+                let _ = channel.wait_close();
+                Ok(())
+            })() {
+                Ok(_) => break,
+                Err(e) => {
+                    counter += 1;
+                    if counter >= 6 {
+                        panic!("Took too many attempts to run command. Last error: {:?}", e);
+                    }
+                }
+            };
+            thread::sleep(std::time::Duration::new(10, 0));
+        }
+        s
+    }
+
+    fn prepare_files() -> (Vec<&'static str>, String) {
+        let mut workload_path = dirs::home_dir().unwrap();
+        workload_path.push("workloads");
+
+        let mut fw_path = workload_path.clone();
+        fw_path.push("hypervisor-fw");
+
+        let mut osdisk_base_path = workload_path.clone();
+        osdisk_base_path.push("clear-29620-cloud.img");
+
+        let osdisk_path = "/tmp/osdisk.img";
+        let cloudinit_path = "/tmp/cloudinit.img";
+
+        fs::copy(osdisk_base_path, osdisk_path).expect("copying of OS source disk image failed");
+
+        let disks = vec![osdisk_path, cloudinit_path];
+
+        (disks, String::from(fw_path.to_str().unwrap()))
+    }
+
+    #[test]
+    fn test_simple_launch() {
+        let handler = thread::spawn(|| {
+            let (disks, fw_path) = prepare_files();
+
+            let vm_config = config::VmConfig::parse(config::VmParams {
+                cpus: "1",
+                memory: "512",
+                kernel: fw_path.as_str(),
+                cmdline: None,
+                disks,
+                rng: "/dev/urandom",
+                net: Some("tap=,mac=,ip=192.168.2.1,mask=255.255.255.0"),
+            })
+            .expect("Failed parsing parameters");
+
+            vmm::boot_kernel(vm_config).expect("Booting kernel failed");
+        });
+
+        thread::sleep(std::time::Duration::new(10, 0));
+        assert_eq!(ssh_command("grep -c processor /proc/cpuinfo").trim(), "1");
+        assert_eq!(
+            ssh_command("grep MemTotal /proc/meminfo").trim(),
+            "MemTotal:         496400 kB"
+        );
+
+        assert!(
+            ssh_command("cat /proc/sys/kernel/random/entropy_avail")
+                .trim()
+                .parse::<u32>()
+                .unwrap()
+                >= 1000
+        );
+
+        ssh_command("sudo reboot");
+
+        handler.join().unwrap();
+    }
+
+    #[test]
+    fn test_simple_launch_again() {
```

Comment:
Any reason why we're running that test twice?

Reply:
This originally came from when the tests were being running via the internals and it was useful to check that running a second test worked (it didn't at some point.) With running via the binary the value is no longer there and I will remove it.